apps/mobile/src/app/account/[accountId]/activity.tsx

```diff
@@ -43,7 +42,7 @@ export default function AccountActivityScreen() {
                 })}
               </Screen.Title>
             }
-            data={activity.value as OnChainActivity[]} // TODO: Unclear why was this cast. Needs clearing up.
+            data={activity.value.filter(activity => activity && 'asset' in activity)}
```

Comment:
Filtering for only `OnChainActivity` to avoid casting. I could probably do this in the Query itself